airbyte-workers/src/main/java/io/airbyte/workers/process/KubePodProcess.java

```diff
@@ -179,7 +230,18 @@ public InputStream getErrorStream() {
 
   @Override
   public int waitFor() throws InterruptedException {
+    // These are closed in the opposite order in which they are created to prevent any resource
+    // conflicts.
     client.resource(podDefinition).waitUntilCondition(this::isTerminal, 10, TimeUnit.DAYS);
+    try {
+      this.stdin.close();
+      this.stdoutServerSocket.close();
+      this.stdout.close();
+      this.executorService.shutdownNow();
+    } catch (IOException e) {
+      e.printStackTrace();
```

Comment:
rethrow as runtime exception?

Reply:
good call. inserted a logging statement and throwing an interrupted exception to respect the method signature.